src/test/java/com/splunk/kafka/connect/SplunkSinkTaskTest.java

```diff
@@ -246,6 +249,91 @@ public void putWithRawAndAck() {
         putWithSuccess(true, true);
     }
 
+    @Test
+    public void checkExtractedTimestamp() {
+        SplunkSinkTask task = new SplunkSinkTask();
+        Collection<SinkRecord> record = createSinkRecords(1,"{\"id\": \"19\",\"host\":\"host-01\",\"source\":\"bu\",\"fields\":{\"hn\":\"hostname1\",\"CLASS\":\"class1\",\"cust_id\":\"000013934\",\"time\": \"Jun 13 2010 23:11:52.454 UTC\",\"category\":\"IFdata\",\"ifname\":\"LoopBack7\",\"IFdata.Bits received\":\"0\",\"IFdata.Bits sent\":\"0\"}");
+        // SinkRecord record = new SinkRecord(new UnitUtil(0).configProfile.getTopics(), 1, null, null, null, "{\"id\": \"19\",\"host\":\"host-01\",\"source\":\"bu\",\"fields\":{\"hn\":\"hostname1\",\"CLASS\":\"class1\",\"cust_id\":\"000013934\",\"time\": \"Jun 13 2010 23:11:52.454 UTC\",\"category\":\"IFdata\",\"ifname\":\"LoopBack7\",\"IFdata.Bits received\":\"0\",\"IFdata.Bits sent\":\"0\"}", 0, 0L, TimestampType.NO_TIMESTAMP_TYPE);
+        UnitUtil uu = new UnitUtil(0);
+        Map<String, String> config = uu.createTaskConfig();
+        config.put(SplunkSinkConnectorConfig.RAW_CONF, String.valueOf(false));
+        config.put(SplunkSinkConnectorConfig.ENABLE_TIMESTAMP_EXTRACTION_CONF, String.valueOf(true));
+        config.put(SplunkSinkConnectorConfig.REGEX_CONF, "\\\"time\\\":\\s*\\\"(?<time>.*?)\"");
+        config.put(SplunkSinkConnectorConfig.TIMESTAMP_FORMAT_CONF, "MMM dd yyyy HH:mm:ss.SSS zzz");
+        HecMock hec = new HecMock(task);
+        hec.setSendReturnResult(HecMock.success);
+        task.setHec(hec);
+        task.start(config);
+        task.put(record);
+       
+        List<EventBatch> batches = hec.getBatches();
+        for (Iterator<EventBatch> iter = batches.listIterator(); iter.hasNext();) {
+            EventBatch ch = iter.next();
+            List<Event> ev = ch.getEvents();
+            Iterator<Event> iter1 = ev.listIterator() ;
+            Event ev1 = iter1.next();
+            Assert.assertEquals(1.276470712454E12+"",ev1.getTime()*1000+"");
```

Comment:
Don't have much experience with Java, but why do we need to `+""` here to compare values?

Reply:
Resolved it here: 
https://github.com/splunk/kafka-connect-splunk/blob/timestamp-parsing/src/test/java/com/splunk/kafka/connect/SplunkSinkTaskTest.java#L272

("" was to convert it to string.Used double precision instead)